storage/remote/storage.go

```diff
@@ -62,37 +63,49 @@ func (s *Storage) ApplyConfig(conf *config.Config) error {
 
 	// Update write queues
 
-	newQueues := []*QueueManager{}
+	newQueues := make([]*QueueManager, 0, len(s.queues))
+	oldQueues := s.queues
 	// TODO: we should only stop & recreate queues which have changes,
 	// as this can be quite disruptive.
 	for i, rwConf := range conf.RemoteWriteConfigs {
-		c, err := NewClient(i, &ClientConfig{
-			URL:              rwConf.URL,
-			Timeout:          rwConf.RemoteTimeout,
-			HTTPClientConfig: rwConf.HTTPClientConfig,
-		})
+		ok := i < len(oldQueues)
+		if ok {
+			qm := oldQueues[i]
+			if reflect.DeepEqual(conf.GlobalConfig.ExternalLabels, qm.externalLabels) &&
```

Comment:
I'm fine with using reflect.DeepEqual in a test, but not in actual production code.  model.LabelSet has a Equal function.